src/sema/expression/assign.rs

```diff
@@ -13,6 +13,22 @@ use solang_parser::diagnostics::Diagnostic;
 use solang_parser::pt;
 use solang_parser::pt::CodeLocation;
 
+///Helper functions
```

Comment:
That's not a useful comment. How about `// ensure immutable are only mutated in the constructor`